zerver/views/streams.py

```diff
@@ -283,22 +290,41 @@ def add_subscriptions_backend(request, user_profile,
             else:
                 stream_msg = "a new stream #**%s**." % created_streams[0].name
             msg = ("%s just created %s" % (user_profile.full_name, stream_msg))
+
+            sender = get_user_profile_by_email(settings.NOTIFICATION_BOT)
+            stream_name = notifications_stream.name
+            topic = 'Streams'
+
             notifications.append(
-                internal_prep_message(user_profile.realm, settings.NOTIFICATION_BOT,
-                                      "stream",
-                                      notifications_stream.name, "Streams", msg))
+                internal_prep_stream_message(
+                    realm=user_profile.realm,
+                    sender=sender,
+                    stream_name=stream_name,
+                    topic=topic,
+                    content=msg))
+
         else:
             msg = ("Hi there!  %s just created a new stream #**%s**."
                    % (user_profile.full_name, created_streams[0].name))
+
+            sender = get_user_profile_by_email(settings.NOTIFICATION_BOT)
             for realm_user_dict in get_active_user_dicts_in_realm(user_profile.realm):
                 # Don't announce to yourself or to people you explicitly added
                 # (who will get the notification above instead).
                 if realm_user_dict['email'] in principals or realm_user_dict['email'] == user_profile.email:
                     continue
-                notifications.append(internal_prep_message(
-                    user_profile.realm, settings.NOTIFICATION_BOT,
-                    "private",
-                    realm_user_dict['email'], "", msg))
+
+                recipient_email = realm_user_dict['email']
+
+                notifications.append(
+                    internal_prep_private_message(
+                        realm=user_profile.realm,
+                        sender=sender,
+                        recipient_email=recipient_email,
+                        content=msg))
+
+    for stream in created_streams:
+        notifications.append(prep_stream_welcome_message(stream))
```

Comment:
(Basically because in the Zephyr or more generally mirror world, creating a stream in Zulip may be actually just creating a reflection of an already-existing thing outside Zulip).